src/averageAges.js

```diff
@@ -55,7 +82,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenList = people
+    .filter(person => {
+      const child = people
+        .find(potentialChild => person.mother === potentialChild.name);
+
+      return child;
+    })
+    .filter(child => onlyWithSon ? (child && child.sex === 'm') : child);
+  const differenceOfAgesList = childrenList.map(child => {
```

Comment:
```suggestion
  const differenceOfAges= childrenList.map(child => {
```

Reply:
ther are no `list`s in the code finally :))